src/Providers/Stripe/Payment/Service.php

```diff
@@ -43,22 +45,16 @@ public function create(
         Token $token,
         int $amount,
         string $currency = 'GBP',
-        string $description = null,
-        array $metadata = []
+        array $metadata = [],
+        $customer = null,
+        string $description = null
     ): Payment {
         switch ($token->type) {
             case StripeToken::CUSTOMER:
-                $paymentMethods = PaymentMethod::all([
-                    'customer' => $token->value,
-                    'type' => 'card',
-                ]);
-
-                /** @var PaymentMethod $paymentMethod */
-                $paymentMethod = $paymentMethods->data[0];
+                $cards = $this->cards()->index($token);
+                $card = $cards[0];
 
-                $paymentMethodToken = new StripeToken($paymentMethod->id);
-
-                if ($paymentMethodToken->type === StripeToken::CARD) {
+                if ($token->isCard($card->uid)) {
```

Comment:
I think @LeeBrooks3 did not like this method and, master version looks differently

Reply:
ah okay my bad I kept the method especially for that but yeah as we have StripeToken we could use it instead of this.